.golangci.yml

```diff
@@ -0,0 +1,153 @@
+    - godox # tool for detection of FIXME, TODO and other comment keywords
+
+# all available settings of specific linters
+linters-settings:
+  errcheck:
+    # report about not checking of errors in type assertions: `a := b.(MyStruct)`;
+    # default is false: such cases aren't reported by default.
+    check-type-assertions: true
+
+  errorlint:
+    # Check whether fmt.Errorf uses the %w verb for formatting errors. See the readme for caveats
+    errorf: true
+    # Check for plain type assertions and type switches
+    asserts: true
+    # Check for plain error comparisons
+    comparison: true
+
+  goconst:
+    # minimal length of string constant, 3 by default
+    min-len: 3
+    # minimal occurrences count to trigger, 3 by default
+    min-occurrences: 2
+
+  makezero:
+    # Allow only slices initialized with a length of zero. Default is false.
+    always: true
+
+  dupl:
+    # tokens count to trigger issue, 150 by default
+    threshold: 100
+
+  forbidigo:
+    # Forbid the following identifiers
+    forbid:
+      - fmt.Print.* # too much log noise
+    # Exclude godoc examples from forbidigo checks.  Default is true.
+    exclude_godoc_examples: true
+
+  gomoddirectives:
+    # Allow local `replace` directives. Default is false.
+    replace-local: false
+
+  gosimple:
+    # Select the Go version to target. The default is '1.13'.
+    go: "1.17"
+
+  misspell:
+    # Correct spellings using locale preferences for US or UK.
+    # Default is to use a neutral variety of English.
+    # Setting locale to US will correct the British spelling of 'colour' to 'color'.
+    locale: US
+    # ignore-words:
+    #   - IdP
+
+  nakedret:
+    # make an issue if func has more lines of code than this setting and it has naked returns; default is 30
+    max-func-lines: 0
+
+  prealloc:
+    # XXX: we don't recommend using this linter before doing performance profiling.
+    # For most programs usage of prealloc will be a premature optimization.
+
+    # Report preallocation suggestions only on simple loops that have no returns/breaks/continues/gotos in them.
+    # True by default.
+    simple: true
+    range-loops: true # Report preallocation suggestions on range loops, true by default
+    for-loops: false # Report preallocation suggestions on for loops, false by default
+
+  nolintlint:
+    # Enable to ensure that nolint directives are all used. Default is true.
+    allow-unused: false
+    # Disable to ensure that nolint directives don't have a leading space. Default is true.
+    allow-leading-space: true
+    # Exclude following linters from requiring an explanation.  Default is [].
+    allow-no-explanation: []
+    # Enable to require an explanation of nonzero length after each nolint directive. Default is false.
+    require-explanation: true
+    # Enable to require nolint directives to mention the specific linter being suppressed. Default is false.
+    require-specific: true
+
+  staticcheck:
```

Comment:
NIT: It would be nice to have the `1.17` to be populated by the `.go-version` file.

Reply:
In the long run it would be nice to have it reading from `.go-version` so we don't need to remember to update this configuration file as we change our Go version.

Because we're most likely running it with mage, I see no issue on adding this automation.

Reply:
@rdner This shouldn't be a lot of effort,  we have already code generation. There is a `make update` task that generates the configuration files from templates and other generated files. That `make update` comment is run on every build, if `git status` detects new files or modified files it will fail the build.

So the workflow would be like this.

- Update `.go-version`
- run `make update`
- Push PR with the updated linter information and the updated go version.
- We have an automated task that does this https://github.com/elastic/fleet-server/pull/1143

But on the other side we will have 1.17 for some time and 1.18 for quite some time, I just hate having multiple files to updates we still have too much and removing the human in the operation is always a good thing and having multiple go repository that just work the same would save us time.

Reply:
@ph This PR already has a ton of comments, I'm not going to implement it here but I can take as the next step.